Updates per review & feedback (#611)
temporalio/documentation: docs/content/how-to-get-the-result-of-a-workflow-execution-in-go.md

To wait on the result of Workflow Execution in the same process that invoked it, call `Get()` on the instance of `WorkflowRun` that is returned by the `ExecuteWorkflow()` call.

```go
  workflowRun, err := c.ExecuteWorkflow(context.Background(), workflowOptions, YourWorkflowDefinition, param)
  if err != nil {
    // ...
  }
  var result YourWorkflowResponse
  err = workflowRun.Get(context.Background(), &result)
  if err != nil {
      // ...
  }
  // ...
}
```

> However, the result of a Workflow Execution can be obtained from a completely different process, all that is needed is the [Workflow Id](#) (a [Run Id](#) is optional if there is more than one closed Workflow Execution with the same Workflow Id).

Reviewer: ```suggestion
However, the result of a Workflow Execution can be obtained from a completely different process.
All that is needed is the [Workflow Id](#).
(A [Run Id](#) is optional if more than one closed Workflow Execution has the same Workflow Id.)
```
